File: Java & Python/trees/pre_order.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
	public ArrayList<Integer> preorderTraversal(TreeNode a) {
		ArrayList<Integer> res = new ArrayList<Integer>();
		Stack<TreeNode> stack = new Stack<>();

		if (a == null)
			return res;
		stack.push(a);
		TreeNode node;

		while (!stack.isEmpty()) {
			node = stack.pop();
			res.add(node.val);

			if (node.right != null)
				stack.push(node.right);
			if (node.left != null)
				stack.push(node.left);
		}
		return res;
	}
}

File: Java & Python/trees/pre_order.py

# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
	# @param A : root node of tree
	# @return a list of integers
	def preorderTraversal(self, A):
		res = []
		stack = []
		if A is None:
			return res
		stack.append(A)
		while len(stack):
			node = stack.pop(0)
			res.append(node.val)
			if node.right != None:
				stack.insert(0,node.right)
			if node.left != None:
				stack.insert(0,node.left)
			
		return res

